proof/refine/ARM/Finalise_R.thy

```diff
@@ -2326,6 +2326,342 @@ lemma replyClear_invs'[wp]:
   apply wpsimp
   sorry
 
+lemma isHeadSome:
+ "isHead (Some h) \<longleftrightarrow> (\<exists>p. h = Head p)"
+  by (clarsimp simp: isHead_def split: reply_next.split)
+
+lemma in_list_refs_of_replies'_E:
+  assumes in_list: "(p, tp) \<in> list_refs_of_replies' s replyPtr"
+  assumes nextR: "\<And>reply. tp = ReplyNext \<Longrightarrow> replyNext reply = Some (Next p)
+                            \<Longrightarrow> replies_of' s replyPtr = Some reply \<Longrightarrow> R"
+  assumes prevR: "\<And>reply. tp = ReplyPrev \<Longrightarrow> replyPrev reply = Some p
+                            \<Longrightarrow> replies_of' s replyPtr = Some reply \<Longrightarrow> R"
+  shows "R"
```

Comment:
This doesn’t seem to be used, either. Do we want to keep it?

Reply:
I think I'll add a comment saying "FIXME RT: unused?"

Reply:
Ok. I guess we can revisit when we finish Refine.